linode/manager
# feat: [M3-8332] - Use new "lish" API instead of "lish_token"

## Description 📝
This change upgrades the backend weblish/glish session creation requests from the older `lish_token` API to a newer `lish` API. This newer API includes support for both core and distributed regions.

## Changes  🔄
List any change relevant to the reviewer.
- This change uses the new `lish` API which returns the full URLs rather than leaving it up to cloud manager to compute. For core regions, this should result in the same URL as cloud manager would have computed.
- The new `lish` API also returns values to use as WebSocket "protocols"; this will currently be an empty list for core regions, but include session information for distributed regions.

## Target release date 🗓️
ASAP

## Preview 📷
There are no user-visible changes.

## How to test 🧪
Create weblish and glish sessions against a host. It should work exactly as before for core regions. In the browser Developer Tools, you should see accesses to the `lish` endpoint and _not_ the `lish_token` endpoint.

### Prerequisites
No special setup is required.

### Reproduction steps
N/A

### Verification steps
See "How to test" above.

## As an Author I have considered 🤔

*Check all that apply*


---
## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**
- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

---
